code/game/objects/radiation.dm

```diff
@@ -0,0 +1,35 @@
+/proc/radiation_pulse(turf/epicenter, heavy_range, light_range, severity, log=0)
+	if(!epicenter) return
+
+	if(!istype(epicenter, /turf))
+		epicenter = get_turf(epicenter.loc)
+
+	if(log)
+		message_admins("Radiation pulse with size ([heavy_range], [light_range]) and severity [severity] in area [epicenter.loc.name] ")
+		log_game("Radiation pulse with size ([heavy_range], [light_range]) and severity [severity] in area [epicenter.loc.name] ")
+
+	if(heavy_range > light_range)
+		light_range = heavy_range
+
+	for(var/atom/T in range(light_range, epicenter))
+		var/distance = get_dist(epicenter, T)
+		if(distance < 0)
+			distance = 0
+		if(distance < heavy_range)
+			T.rad_act(severity)
+		else if(distance == heavy_range)
+			if(prob(50))
+				T.rad_act(severity)
+			else
+				T.rad_act(severity / 2)
```

Comment:
We should make the whole "multiply by the reciprocal" thing part of the code standards page.